MapboxVisionTests/RecordCoordinatorTests.swift

```diff
@@ -72,6 +70,69 @@ final class RecordCoordinatorTests: XCTestCase {
         XCTAssert(directoryExists(at: recordingPath), "Recording should be saved at \(recordingPath) after recording is stopped")
     }
 
+    class TestClipRequest {
+        let start: Float
+        let end: Float
+
+        init(start: Float, end: Float) {
+            self.start = start
+            self.end = end
+        }
+    }
+
+    func testVideoClipping(withRequests requests: [TestClipRequest], _ message: String = "", line: UInt = #line) {
+        let expectation = XCTestExpectation(description: "Wait for a big enough amount of time for record processing queue to finish its job")
+        let videoSource = FakeVideoSource()
+        videoSource.add(observer: coordinator)
+        videoSource.start()
+        coordinator.startRecording(referenceTime: 0, videoSettings: videoSettings) {}
+        _ = Timer.scheduledTimer(withTimeInterval: 4.0, repeats: false) { _ in
+            print("Make \(requests.count) clip\(requests.count == 1 ? "" : "s")")
+            requests.forEach {
+                self.coordinator.makeClip(from: $0.start, to: $0.end)
+            }
+
+            _ = Timer.scheduledTimer(withTimeInterval: 1.0, repeats: false) { _ in
+                self.coordinator.stopRecording()
+
+                _ = Timer.scheduledTimer(withTimeInterval: 1.0, repeats: false) { _ in
+                    let path = "\(DocumentsLocation.currentRecording.path)"
+                    let directoryContent = (try? FileManager.default.contentsOfDirectory(at: URL(string: path)!, includingPropertiesForKeys: nil, options: []))
+
+                    if let directoryContent = directoryContent {
+                        for file in directoryContent {
+                            print("File \(file)")
+                            if file.hasDirectoryPath {
+                                let subDirectoryContent = (try? FileManager.default.contentsOfDirectory(at: file, includingPropertiesForKeys: nil, options: []))
+                                if let subDirectoryContent = subDirectoryContent {
+                                    let numberOfVideos = subDirectoryContent.filter {
+                                        $0.pathExtension == "mp4"
```

Comment:
Video file type can be taken from `videoSettings.fileExtension`.

Reply:
changed